atom/browser/ui/file_dialog.h

```diff
@@ -33,11 +33,27 @@ enum FileDialogProperty {
   FILE_DIALOG_TREAT_PACKAGE_APP_AS_DIRECTORY = 1 << 7,
 };
 
+#if defined(MAS_BUILD)
 typedef base::Callback<void(
-    bool result, const std::vector<base::FilePath>& paths)> OpenDialogCallback;
+    bool result,
+    const std::vector<base::FilePath>& paths,
+    const std::vector<std::string>& bookmarkData)> OpenDialogCallback;
 
 typedef base::Callback<void(
-    bool result, const base::FilePath& path)> SaveDialogCallback;
+    bool result,
+    const base::FilePath& path,
+    const std::string& bookmarkData)> SaveDialogCallback;
+#else
+typedef base::Callback<void(
+    bool result,
+    const std::vector<base::FilePath>& paths)> OpenDialogCallback;
+
+typedef base::Callback<void(
+    bool result,
+    const base::FilePath& path)> SaveDialogCallback;
+#endif
+
```

Comment:
@alexeykuzmin  Not sure if this is the best way to go about changing `OpenDialogCallback` and `SaveDialogCallback`?
If you have any other suggestions let me know.

Reply:
Hmmm if we conditionally change this here on `MAS_BUILD`s, then lines 56 and 77 of `atom/browser/web_dialog_helper.cc` also need to reflect that (since they use the same call signature).